### ISSUE TITLE
Unexpected VM Charge when using Virtual nodes

### ISSUE THREAD

**Issue by ianadavies** [virtual-kubelet/azure-aci#154]
I am using Kubernetes in Azure with Virtual Nodes this is a plugin that creates virtual nodes using Azure Container Instances.

The instructions to set this up require creating a AKSNet/AKSSubnet which seems to automatically come along with A VMSS called something like. aks-control-xxx-vmss I followed the instruction on the link below.

https://docs.microsoft.com/en-us/azure/aks/virtual-nodes-cli

This comes with a single instance VM that I am being charged full price for regardless of container instances I create and I am charged extra for every container instance I provision onto my virtual node pool even if they should fit on just 1 VM. These resources do not seem to be related.

I am currently having this out as unexpected billing with Microsoft but the process has been very slow so I am reverting to here to find out if anyone else has had this experience?

The main questions I have are:

- Can I use Azure Container Instances without the VMSS?
- If not can I somehow make this VM visible to my cluster so I can at least use it to provision containers onto and get some value out of it?
- Have I just done something wrong?

NB: 
This is not my control node that is a B2s which I can see my system containers running on.
If I stop this single instance VM in my VMSS then my ACI node becomes unavailable.
This seems to defeat the object of using ACI

Any advice would be a great help.

**Comment by feiskyer** [comment]
this is expected as there needs an existing Kubernetes cluster to run virtual kubelet components.

If you don't want to use virtual node, you could follow https://docs.microsoft.com/en-us/azure/container-instances/ to use Azure ACI APIs and CLI to manage ACI instances.

**Comment by ianadavies** [comment]
@feiskyer, Ok I think your answer has raised more questions than it answered. 

- I'm confused, why can't these components run on my existing control node? 
- If that is not possible is there a reason this node needs to be the same spec as my Scaleset, e.g if I want nodes that are 
   Dvs4 then so is this control node which is expensive. 
- Can I run my own custom containers on this node to make better use of the resources?
- I thought the whole point of virtual kubelets (this project) was to take advantage of Azure Container Instances within 
   Kubernetes? I am even more confused.

Thanks 


Ian

**Comment by feiskyer** [comment]
1. It works of course on your existing kubernetes cluster. Please follow https://github.com/virtual-kubelet/azure-aci#deployment-of-the-aci-provider-in-your-cluster to deploy via helm.
2. No such requirement
3. yes, of course
4. Refer https://github.com/virtual-kubelet/virtual-kubelet#how-it-works for how virtual kubelet works.

**Comment by ianadavies** [comment]
It's ok got it, thanks @feiskyer

A long story, but put simply I got confused with the AKS Subnets that I created when following the instructions for virtual kubelet and the VMSS hanging off that, I thought that was effectively the bridge to ACI container instances.

The confusion was compounded by the fact my control node was a B2 and the VMSS was DsV4, so it looked like 2 different VMSS's, it was just that the control node model had not been updated but I was still getting charged for DsV4.

In case anyone else has this confusion the VMSS has nothing to do with the virtual nodes.
